### DISCUSSION TITLE
Question about the current state of development

### DISCUSSION THREAD

**Discussion by mxpand** [SynthstromAudible/DelugeFirmware#2684] [General]
Cheers! Mathias here from mxpand (overlays). I have a question for all developers and the current status of the beta releases.

Many shortcuts have been duplicated for one and the same function (or very similar functions). Depending on which view you are currently in (Grid or Row, Clip or Song), you can now find the shortcuts duplicated in different “places” with different key combinations.

Hence my question: Is this doubling of the shortcuts only temporary or should it remain as it is at the moment in the final 1.2 release?

Here are a few examples to give you a better understanding:
- “Performance View” in Row View = Keyboard Button... in Grid View => pink Sidebar Button
- "Create Audio Clip" is in Row View => press Select... in Grid View => press “Cross Screen” button
- "Transpose Semitone" is in Clip View => Shift + press/turn vertical encoder... in Song View => press/turn encoder (without Shift)
- “Arm Clips” is in Row View => Rec + Mute button... in Grid View => Rec +  corresponding clip button
- “Clip Mode” (Fill, Once...) is in Row View = Mute Button + Select... in Grid View => corresponding Clip Button + Select
- ...

From a user perspective, it is very confusing (at least for me) that I have to remember so many shortcuts twice - for one and the same (or similar) function. So I'd rather ask now before I start duplicating all the shortcuts on a new overlay. Space is generally very limited ;)

Thanks for the answer whether this will remain so in the final 1.2 release.

**Comment by seangoodvibes** [comment]
HI Mathias (@mxpand),

Welcome.

I can answer some of your questions:

- Performance View: they are separate shortcuts because the Keyboard button in Grid View was being reserved for opening up a keyboard at the bottom of the grid. In row view there is no where we could place a "pink pad" to open / close performance view.

- Create an Audio Clip: this has now changed in the nightly and beta firmware. When you create a new clip in a new track, it will open the clip type menu where you choose the clip type to be created. To create an audio clip in this menu, you scroll up to audio clip and press select.

- Convert an Audio Clip: this has also been changed in the nightly and beta firmware. To convert an instrument clip to an audio clip, you select that instrument clip and press on the select encoder. It will open a Clip Settings menu. The first option at the top will be "Convert to Audio" which you can select by pressing select.

- Arm Clips: in row view you have multiple points of contact to access to modify clip settings (e.g. the sequence and the mute pad). In grid view you only have one point of contact - the clip pad. Hence why green mode and blue mode were added initially so that you would have two points of contact to edit clip settings. There is really not much we can do here to align them as we are not going to be changing row view anymore so that legacy users do not have to learn anything new.

- Clip Mode: same reason as Arm Clips above. We are limited by the real-estate that a clip has in grid view. One pad vs several pads in row view.

- Transpose: In Song View, transpose is a master transpose and it's different than the feature in clips which was changed in a recent community firmware release. In clip view:

> The command `SHIFT` + hold and turn `▼︎▲︎` inside a clip was causing an unexpected behavior in which all other clips in the song were also transposed. This has been fixed by changing this command to a "Vertical nudge" command, based on current clip display (either in scale or non-scale mode). This saves user from the need to "zoom out, copy all notes, scroll up or down, and paste all notes" to nudge notes vertically.
>     - If the clip is in scale mode, all the notes are shifted up or down by one step in the scale.
>     - If the clip is not in scale mode, all the notes are shifted up or down by one semitone.
>     - Note: the other command for octave transposition, that is, hold and turn `▼︎▲︎`, keeps working in the same way, by
>       nudging notes by one octave, regardless of the clip scale mode.

The song view master transpose allows you to transpose by a customizable number of semitones. `SHIFT` + hold and turn `▼︎▲︎` adjusts the number of semitones that hold and turn `▼︎▲︎` (without shift) will transpose by.

Lastly, I will add that not everything is 100% set in stone. We are reviewing how things are being implemented to ensure consistency. The cross screen example is a good one and we changed that recently after realizing it created too much UI inconsistency.

Hope this helps.

Regards,

Sean

**Comment by seangoodvibes** [comment]
@mxpand upon further discussion with @ok-reza and @m-m-adams 

Our view is that we are not opposed to making further changes to bring the song grid view into closer alignment with the song row view.

What I think this means is that I'm not sure we're ready to commit to a layout for your overlay just yet.
 
Maybe certain things could be omitted from the first community firmware overlay.

We are definitely open to discussing this more with you. If you want you can join our discord and we can have more conversions on there about it.

https://discord.gg/WVyEYxmz

We also have a weekly call on Discord on Sunday at 3:30 PM eastern time where we discuss development matters.

Regards,

Sean

**Comment by m-m-adams** [comment]
Hey Mathias! Just want to share a picture of my own mxpand overlay as a reference - I've added labels for a couple of the new features that I find helpful (fill, automation, wavefolding, filter morph/drive). I'm planning to expand fold into a full wave shaping section on those 3 available shortcut pads, probably something like [FOLD, SHAPE, ASYMMETRY] top to bottom. I'm also planning to put another sequence control next to direction, something like rate/probability. 

This is obviously up to you but one of our big efforts right now is making everything accessible in the menu to remove the need to remember shortcuts for things like LED brightness. All note related settings will be in a note menu accessed via [NOTE + SELECT], clip settings in [CLIP PAD + SELECT] from grid view or just [SELECT] in a clip, song settings from [SELECT] in song or arranger. 

![IMG_2635](https://github.com/user-attachments/assets/4d9b47a5-cb36-4f20-8aed-66ffadcd8ce9)


Also yeah as Sean mentioned we're keeping rows mode as unchanged as possible for people who are used to it. We see the grid as the main view going forward and the fact that each clip gets one pad vs. 3 (row pads, mute, audition) means it needs different interaction patterns. 

We're trying to keep it as consistent as we can within that by putting the main interaction patterns (select/launch, arm, learn midi) on the clip pad and moving other things to the clip menu but it's a bit of a work in progress (particularly since none of us are getting paid to work on this)

**Comment by mxpand** [comment]
Many thanks to both of you (@seangoodvibes and @m-m-adams) for the very detailed answers, I really very much appreciate it!


> What I think this means is that I'm not sure we're ready to commit to a layout for your overlay just yet.

Ok, if features are still going to change, I probably started designing the community 1.2 overlay way too early. I will wait until only bugs are fixed (and nothing else will change).

> We are definitely open to discussing this more with you. If you want you can join our discord and we can have more conversions on there about it.

Thanks. It's probably the better place to discuss things, as there seem to be more people active there.

> Just want to share a picture of my own mxpand overlay as a reference - I've added labels for a couple of the new features that I find helpful (fill, automation, wavefolding, filter morph/drive). I'm planning to expand fold into a full wave shaping...

Hey, thanks for the photo, looks great. :) I have already integrated the shortcuts you labeled into the design. I'm very curious to see what else is coming!

> This is obviously up to you but one of our big efforts right now is making everything accessible in the menu to remove the need to remember shortcuts for things like LED brightness. 

Ah, I see. I can only speak for myself, of course, but menu diving (and reading displays) is not really my thing. So I hope that there will continue to be shortcuts (also for new functions).

**Reply by seangoodvibes** [reply]
Absolutely there will continue to be shortcuts for new functions! We just want to maximize the potential of the OLED display as well.

We've even discussed adding a second layer of shortcuts to the grid. E.g. we have Envelope 1 and Envelope 2 on the grid. We envision adding Envelope 3 and 4 which would be accessed respectively by pressing the Envelope 1 and 2 shortcuts twice.

**Reply by mxpand** [reply]
Ok, that's great!

**Comment by mxpand** [comment]
So that I can start the final design of the 1.2 overlay in good time (but not too early), I have 2 important questions:

- Is there an easy way to know that after a certain point of a beta version ONLY bugs will be fixed - and 100% no new features (or changes) are coming? Is this announced somewhere, or can it be recognized by the name of a beta?
- I fully realize that software is only “finished” when it is finished. But is there perhaps already a very rough idea of when the official 1.2 version will be released? Maybe the end of the year, or next year (or even an approximate month)?

That would help a lot. Thank you very much.

**Reply by m-m-adams** [reply]
Oh and once we call it a beta we're not adding features, once it's an RC we're not changing anything. Time wise that'll hopefully be in October

**Reply by mxpand** [reply]
Thank you very much for the information, that helps me a lot. Then I will wait until the RC version before I continue.

**Comment by m-m-adams** [comment]
Yeah definitely keeping the shortcuts! I just feel having menus is important since it lets you do things even if you forget them

**Reply by mxpand** [reply]
That's also good to know. I assume that this new feature (and others) will not come before c1.3, as the current status is already “beta”.
